File: Dijkstra/cpp/node.cpp
#include "node.h"

node_t::node_t(int index, int dist, node_t* prev): index(index), dist(dist), prev(prev) {};

node_t::node_t(){};

int node_t::get_index(){
    return index;
}

int node_t::get_dist(){
    return dist;
}

node_t* node_t::get_prev(){
    return prev;
}

File: Dijkstra/Java/Node.java
package Dijkstra.Java;

public class Node {
    int dist;
    int index;
    char name;
    Node prev;

    Node(int name, int dist, Node prev){
        this.name = (char)(name + 'a');
        this.index = name;
        this.dist = dist;
        this.prev = prev;
    }

    @Override
    public boolean equals(Object obj) {
        Node o = (Node) obj;
        return o.name == this.name;
    }

    @Override
    public String toString() {
        if(this.prev == null){
            return name + ":" + dist + ":null"; 
        }
        return name + ":" + dist + ":" + prev.name; 
    }
}
